Dalamud/Game/Network/Internal/NetworkHandlers.cs

```diff
@@ -544,7 +556,8 @@ private void CustomTalkReceiveResponseDetour(nuint a1, ushort eventId, byte resp
     {
         try
         {
-            if (eventId == 7 && responseId == 8)
+            // Event ID 0 covers the crystarium, 7 covers all other cities
+            if (eventId is 7 or 0 && responseId == 8)
```

Comment:
Does eventID 0 still follow the same layout?

Also, for @karashiiro - what would you think if we exposed a different event for tax information that could update per-city? This way, we could send tax updates far more frequently than via Retainer Vocates, and probably overload this to send a batch.

Reply:
Context: https://discord.com/channels/581875019861328007/653504487352303619/1246347031786684416 (Some more context higher up in the conversation)

Reply:
> what would you think if we exposed a different event for tax information that could update per-city?

SGTM, the API should actually support this today - the [request schema](https://github.com/Universalis-FFXIV/Universalis/blob/02b776cfb445152d33974a21af5c65c05b3a7e03/src/Universalis.Application/Uploads/Schema/MarketTaxRates.cs) has each city marked as optional, and the [handler logic](https://github.com/Universalis-FFXIV/Universalis/blob/5a98ccd1e2bf772ce114e60bf3799f5967d2eafb/src/Universalis.Application/Uploads/Behaviors/TaxRatesUploadBehavior.cs#L36-L49) reuses the existing data for any missing fields (I could actually make this atomic but that's a server-only change so irrelevant from Dalamud's perspective).

> and probably overload this to send a batch.

Not clear on this part, can you clarify this?

Reply:
>> and probably overload this to send a batch.
> 
> Not clear on this part, can you clarify this?

If the user goes to a retainer vocate, we capture everything and sent it in one batch to your API. If the user just summons their retainer, we'd just submit the one tax rate we currently see.

For *our* API, we'd probably do something similar where we'd either call that event _n_ times or define a `List<MarketTaxRecord>` for the return type.

Reply:
> Does eventID 0 still follow the same layout?
> 
> Also, for @karashiiro - what would you think if we exposed a different event for tax information that could update per-city? This way, we could send tax updates far more frequently than via Retainer Vocates, and probably overload this to send a batch.

From my testing, yes it does

In regards to your point about per city, I agree, I'm not sure I'll make it happen in this particular PR but it should be a matter of finding the correct packet handler and hooking it in the same way.